app/helpers/sessions_helper.rb

```diff
@@ -4,15 +4,60 @@ def log_in user
   end
 
   def current_user
-    @current_user ||= User.find_by id: session[:user_id]
+    if (user_id = session[:user_id])
+      @current_user ||= find_user_from_session(user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      @current_user ||= find_user_from_cookies(user_id)
+    end
+    @current_user
   end
 
   def logged_in?
     current_user.present?
   end
 
   def log_out
+    forget current_user
     session.delete :user_id
     @current_user = nil
   end
+
+  def remember_signin user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def forget user
+    user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
+  end
+
+  def session_signin user
+    forget user
+    session[:session_token] = user.create_session_token
+    session[:user_id] = user.id
+  end
+
+  private
+
+  def find_user_from_session user_id
+    user = User.find_by(id: user_id)
+    return nil if user.nil?
+    if session[:session_token].present? &&
+       !user.authenticated_with_session?(session[:session_token])
+      return nil
+    end
+
+    user
```

Comment:
```suggestion
    return unless user = User.find_by(id: user_id)
    return unless user.authenticated_with_session?(session[:session_token])
       
    user
```
Viết gọn lại được e nhé, return k có giá trị chính là return nil rồi
Mà chỗ này chỉ cần user không authenticated_with_session thì là return false rồi pk, k cần check session[:session_token] có present k